draft-ietf-ppm-dap.md

```diff
@@ -1305,50 +1493,89 @@ given batch interval, denoted `batch_interval`, decryption works as follows:
+share (`0x02` for the leader and `0x03` for the helper). The value of
+`batch_selector` is computed by the Collector from its query and the response to
+its query:
+
+* For time-interval tasks, the batch selector is the batch interval specified in
+  the query.
 
-### Validating Batch Parameters {#batch-parameter-validation}
+* For fixed-size tasks, the batch selector is the batch ID assigned sent in the
+  response.
+
+### Batch Validation {#batch-validation}
 
 Before an aggregator responds to a collect request or aggregate-share request,
 it must first check that the request does not violate the parameters associated
 with the DAP task. It does so as described here.
 
-First the aggregator checks that the request's batch interval respects the
-boundaries defined by the DAP task's parameters. Namely, it checks that both
-`batch_interval.start` and `batch_interval.duration` are divisible by
-`min_batch_duration` and that `batch_interval.duration >= min_batch_duration`.
-Unless both these conditions are true, the aggregator MUST abort and alert its
-peer with error "batchInvalid".
-
-Next, the aggregator checks that the request respects the generic privacy
-parameters of the DAP task. Let `X` denote the set of reports for which the
-aggregator has recovered a valid output share and which fall in the batch
-interval of the request.
-
-* If `len(X) < min_batch_size`, then the aggregator MUST abort and alert its
-  peer with "insufficientBatchSize".
-* The aggregator keeps track of the number of times each report was added to the
-  batch of an AggregateShareReq. If any report in `X` was added to at least
-  `max_batch_lifetime` previous batches, then the aggregator MUST abort and
-  alert the peer with "batchLifetimeExceeded".
-
-Finally, the aggregator checks that the batch interval defined by the collect
-request satisfies one of the conditions:
-
-1. The batch interval does not overlap with the batch interval of any prior
-   completed collect requests.
-1. The batch interval, including its start and duration values, match a prior
-   completed collect request.
+First the aggregator checks that the request respects any batch "boundaries"
+determined by the query configuration:
+
+* For time-interval tasks, the query configuration specifies the "minimum batch
+  duration", which is determined by `time_precision`. For the `batch_interval`
+  included with the query, the Aggregator checks that both
+  `batch_interval.start` and `batch_interval.duration` are divisible by
+  `time_precision` and that `batch_interval.duration >= time_precision`.
+
+* For fixed-size tasks, the batch is assocaited with a "batch ID" selected by
+  the Leader. Thus the Aggregator needs to check that the query is associated
+  with a known batch ID:
+
+    * If the query contained in the CollectReq has type `by-batch-id`, the
+      Leader checks that the provided batch ID corresponds to a batch ID it
+      returned in a previous CollectResp for the task.
```

Comment:
Why would the Collector not be able to opportunistically query by batch ID? What if the IDs were monotonically increasing and the Collector just wanted to iterate over the batches?

Reply:
We kicked this around a bit. It seems like this could get tricky. For example, there would need to be a mechanism to ensure that that queries are issued in order: Imagine the collector picks batch id = 10,000, then batch id = 1.

On balance, explicit batch IDs picked by the leader seemed simpler.